File: bookglebookgle-server/src/main/java/com/example/bookglebookgleserver/group/dto/GroupDetailResponse.java
package com.example.bookglebookgleserver.group.dto;

import io.swagger.v3.oas.annotations.media.Schema;
import java.util.List;

@Schema(description = "모임방 상세 조회 응답 DTO")
public record GroupDetailResponse(
        @Schema(description = "모임방 이름", example = "문학 읽기 모임")
        String roomTitle,
        @Schema(description = "카테고리", example = "NOVEL")
        String category,
        @Schema(description = "모임 스케줄", example = "매주 금요일 오후 8시")
        String schedule,
        @Schema(description = "현재 참여 인원 수", example = "5")
        int memberCount,
        @Schema(description = "최대 인원 수", example = "8")
        int maxMemberCount,
        @Schema(description = "모임 설명", example = "함께 문학작품을 읽고 토론하는 모임입니다.")
        String description,
        @Schema(description = "모임 대표 사진 URL (현재는 null)", example = "null")
        String photoUrl,
        @Schema(description = "호스트 여부(요청 사용자 기준)", example = "true")
        boolean isHost,
        @Schema(description = "최소 요구 평점", example = "3")
        int minRequiredRating,

        // ✅ 추가
        @Schema(description = "PDF 총 페이지 수", example = "120")
        int pageCount,
        @Schema(description = "멤버 상세 리스트")
        List<GroupMemberDetailDto> members
) {}

File: bookglebookgle-android/app/src/main/java/com/ssafy/bookglebookgle/entity/GroupDetailResponse.kt
package com.ssafy.bookglebookgle.entity

data class GroupDetailResponse(
    val roomTitle: String,
    val category: String,
    val schedule: String,
    val memberCount: Int,
    val maxMemberCount: Int,
    val description: String,
    val isHost: Boolean,
    val minRequiredRating: Int,
    val pageCount: Int,
    val members: List<GroupMemberDetailDto>,
    val allMembersCompleted: Boolean
)

data class GroupMemberDetailDto(
    val userId: Long,
    val userNickName: String,
    val profileColor: String?,
    val maxReadPage: Int,   // 서버: 0-based
    val progressPercent: Int,
    val isHost: Boolean,
    val ratingSubmitted: Boolean,
    val profileImageUrl: String?,
    val ratedUserIds : List<Long>
)

data class GroupDetail(
    val roomTitle: String,
    val category: String,
    val schedule: String,
    val memberCount: Int,
    val maxMemberCount: Int,
    val description: String,
    val isHost: Boolean,
    val isCompleted: Boolean,          // 모임 종료 여부
    val minRequiredRating: Int,
    val pageCount: Int,
    val members: List<GroupMember>,
)

data class GroupMember(
    val userId: Long,
    val userNickName: String,
    val profileColor: String?,
    val lastPageRead: Int,     // 서버 0-based 유지
    val progressPercent: Int,
    val isHost: Boolean,
    val hasRated: Boolean,
    val profileImageUrl: String?,
    val ratedUserIds : List<Long>
)

fun GroupDetailResponse.toDomain(): GroupDetail = GroupDetail(
    roomTitle = roomTitle,
    category = category,
    schedule = schedule,
    memberCount = memberCount,
    maxMemberCount = maxMemberCount,
    description = description,
    isCompleted = allMembersCompleted,
    minRequiredRating = minRequiredRating,
    isHost = isHost,
    pageCount = pageCount,
    members = members.map { it.toDomain() }
)

fun GroupMemberDetailDto.toDomain(): GroupMember = GroupMember(
    userId = userId,
    userNickName = userNickName,
    profileColor = profileColor,
    lastPageRead = maxReadPage,
    progressPercent = progressPercent,
    isHost = isHost,
    hasRated = ratingSubmitted,   // ★ 서버 필드명은 isCompleted지만 의미는 "평가 완료"
    profileImageUrl = profileImageUrl,
    ratedUserIds = ratedUserIds

)